### ISSUE TITLE
Using Dask serialization in more cases

### ISSUE THREAD

**Issue by jakirkham** [dask/distributed#3946]
Looking through the code we can find a few examples of [functions with arguments and keyword arguments serialized with cloudpickle]( https://blog.dask.org/2019/06/20/load-image-data ). AFAICT this is used in [the `Client`]( https://github.com/dask/distributed/blob/5172678da28130ef7f4d86394d30566abc99328e/distributed/client.py#L2401-L2411 ) and [the `Worker`s]( https://github.com/dask/distributed/blob/df4206365bbb26196f7349b89bc7eccf3cc5de80/distributed/worker.py#L3278-L3292 ). Though maybe it is used in other places as well. In these cases it would be nice to rely on the serialization mechanisms of the underlying `Comm` involved instead of simply pickling.

**Comment by mrocklin** [comment]
I don't have any objection here, other than code complexity.  In practice functions like `run` are only really used for debugging, and so this hasn't come up with users.  If there is an easy way to make this cleaner though then great.

**Comment by jakirkham** [comment]
Yeah I started playing with an idea in PR ( https://github.com/dask/distributed/pull/3948 ). Need to test it a bit more still.

**Comment by dhirschfeld** [comment]
Is this related to:
* https://github.com/dask/distributed/issues/2110
* https://github.com/dask/distributed/issues/2953

**Comment by jakirkham** [comment]
I don't think so. Though agree those would be nice to solve as well.
